Kernel/Scheduler.cpp

```diff
@@ -320,9 +320,9 @@ bool Scheduler::pick_next()
     ASSERT(s_active);
 
     if (!Thread::current) {
-        // XXX: The first ever context_switch() goes to the idle process.
+        // XXX: The first ever context switch goes to the idle process.
         //      This to setup a reliable place we can return to.
-        return context_switch(*g_colonel);
+        return g_colonel;
```

Comment:
Maybe colonel could be `Thread::colonel` too. Or maybe it's a bad idea.

Reply:
Considering it's global and not static I'd say it's probably a good idea to move it into a namespace. It'd probably make more sense to have it as `Kernel::colonel` considering it's _the_ Kernel specific thread though.

Reply:
It's `Kernel::g_colonel` already, isn't it?

Reply:
Yeah it is. IMO the `thread` should be as well, though I'm not too sure how the namespaces are split. My first intuition to find it would definitely be in `Kernel::` though.

Reply:
This sounds good, but I'd prefer to do it in a seperate PR.